tests/builder/axes_test.py

```diff
@@ -305,3 +305,44 @@ def test_single_master_default_weight_400(ufo_module):
 
     assert len(font2.masters) == 1
     assert font2.masters[0].weightValue == 400
+
+
+def test_axis_mapping(ufo_module):
+    font = to_glyphs(
+        [ufo_module.Font(), ufo_module.Font(), ufo_module.Font(), ufo_module.Font()]
+    )
+    font.masters[0].weightValue = 0
+    font.masters[0].widthValue = 100
+    font.masters[1].weightValue = 1000
+    font.masters[1].widthValue = 100
+
+    font.masters[2].weightValue = 0
+    font.masters[2].widthValue = 75
+    font.masters[3].weightValue = 1000
+    font.masters[3].widthValue = 75
+
+    wght_mapping = [(100, 0), (400, 350), (900, 1000)]
+    wdth_mapping = [(75, 75), (100, 100)]
+
+    axis_mappings = {
+        "wght": {k: v for k, v in wght_mapping},
+        "wdth": {k: v for k, v in wdth_mapping},
+    }
+
+    font.customParameters["Axis Mappings"] = axis_mappings
+    doc = to_designspace(font, ufo_module=ufo_module)
+
+    assert doc.axes[0].name == "Weight"
+    assert doc.axes[0].minimum == 100
+    assert doc.axes[0].maximum == 900
+    assert doc.axes[0].map == wght_mapping
+
+    assert doc.axes[1].name == "Width"
+    assert doc.axes[1].minimum == 75
+    assert doc.axes[1].maximum == 100
+    # No axis map needed for Width because user and design coords are identical
+    assert doc.axes[1].map != wdth_mapping
+    assert doc.axes[1].map == []
+
+    font = to_glyphs(doc)
+    assert font.customParameters["Axis Mappings"] == axis_mappings
```

Comment:
are we round-tripping the "Axis Mappings" custom parameter when we go from designspace to .glyphs? We should, but I don't see where this is happening...

Reply:
Yes, I'm round-tripping .

I'm converting the designspace to glyphs and then I'm checking if the Axis Mappings customParameter is equal to the axis_mappings var which is defined on line 327.

Reply:
yes but where in the lib modules this happens? I mean, the setting of customParameters["Axis Mappings"] to the content of a designspace.axes maps? I don't see that in your non-test code

Reply:
We now have three methods to create the mapping. Each method only produces a variable called `mapping`. It doesn't do the assignment. I create this variable in line 164:

`mapping = {float(k): v for k, v in custom_mapping[axis.tag].items()}`

The mapping variable is then converted to a designspace map in line 233:

`axis.map = sorted(mapping.items())`

Reply:
that's the code that sets the DS axes maps from the Glyphs' custom parameter. I am talking about the reverse code, that sets the Glyphs custom parameter from the DesignSpace mappings, when one coverts a DS to .glyphs file.

Reply:
I'm pretty sure `glyphsLib.builder.custom_params` is able to convert customParams to ufo and back again without fault. I could be wrong here but in all my tests, it worked just fine. Just now I made a new glyphs file with an unknown customParam and it has been preserved in a roundtrip.

Reply:
Ah.. I see what's happening. The designspace file `lib` or the master UFOs' `lib.plist` is storing the custom parameter and that explains why you can roundtrip it back. 

What I was looking for is the piece of code (there isn't one currently) that given a designspace file with some axes map elements (but no Glyphs-specific private key in its libs) would construct an "Axis Mappings" custom parameter out of it when creating the GSFont object out of the DS.
We need to think whether we want to do that or not. I would think this would be desirable. /cc @madig @belluzj

Reply:
in any case, this is not clear in @m4rc1e's tests. By looking at the tests only one would think that a new Axis Mappings is created from the designspace axes maps only, whereas what's happening is it is simply being restored. Maybe Marc you could add a comment to clarify that.

Reply:
Sure thing.

Reply:
Yes I think it's very desirable that the "axis mapping" (general term, not implementation-related) is stored as a <map> element under each <axis> in the Designspace file, and actually only there (not as a lib key) so that the "axis mapping" information on each side (Designspace or Glyphs.app) is stored in that side's native format for that information.

Reply:
Instead of leaving a comment, I've decided to implement this. I've removed Jany's old commented out code and replaced it with a simple dict comprehension.